src/components/MapView/Analyser/index.tsx

```diff
@@ -187,8 +195,47 @@ function Analyser({ classes }: AnalyserProps) {
     };
 
     await dispatch(requestAndStoreAnalysis(params));
+    // eslint-disable-next-line fp/no-mutating-methods
+    history.replace(
+      `?hazardLayerId=${hazardLayerId}&baselineLayerId=${baselineLayerId}&date=${selectedDate}&statistic=${statistic}&aboveThreshold=${aboveThreshold}&belowThreshold=${belowThreshold}`,
+    );
   };
 
+  useEffect(() => {
+    const {
+      hazardLayerParamId,
+      baselineLayerParamId,
+      selectedParamDate,
+      statisticParam,
+      aboveThresholdParam,
+      belowThresholdParam,
+    } = extractPropsFromURL(history.location.search);
+
+    if (!hazardLayerParamId || !baselineLayerParamId || !statisticParam) {
+      return;
+    }
+
+    if (isLayerKey(hazardLayerParamId)) {
+      setHazardLayerId(hazardLayerParamId);
+    }
+    if (isLayerKey(baselineLayerParamId)) {
+      setBaselineLayerId(baselineLayerParamId);
+    }
+    if (isStatistic(statisticParam)) {
+      setStatistic(statisticParam);
+    }
+
+    setSelectedDate(selectedParamDate);
+    setAboveThreshold(aboveThresholdParam);
+    setBelowThreshold(belowThresholdParam);
+
+    // Avoid Running Analyser if boundary Layer data is Null
+    if (hazardLayerId && baselineLayerId && selectedDate) {
+      runAnalyser();
+    }
+    // eslint-disable-next-line react-hooks/exhaustive-deps
```

Comment:
You should be able to remove this, and use all dependencies ESLint asks you to include. Let me know if you face issues with this :+1: (we want this code to be written in a way that's independent of the current state of the application, so it remains stable. This eslint rule will help ensure this.)

Reply:
In that case, I'll have to add `hazardLayerId` `baselineLayerId` `selectedDate` etc as well which will cause some issues. (If you do that, you won't be able to change the Analysis from UI, it will go into some infinity loop)